honestcash-v2/src/app/store/wallet/wallet.actions.ts

```diff
@@ -8,10 +8,7 @@ export enum WalletActionTypes {
 
 export class WalletSetup implements Action {
   readonly type = WalletActionTypes.WALLET_SETUP;
-  constructor(public payload: {
-    mnemonicEncrypted: string;
-    password: string;
-  }) {}
+  constructor(public payload: IAuthRequestSuccessResponse) {}
```

Comment:
types do not match.

IAuthRequestSuccessResponse can be casted, but this is not done in function declarations but in runtime conversions etc.